board/flags-config.cmake

```diff
@@ -0,0 +1,37 @@
+# Flags for each supported board
+
+function(flags BOARD_NAME)
+  # Set all minimization flags for all platforms
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Os -fdata-sections -ffunction-sections -fomit-frame-pointer -fno-unwind-tables -fno-asynchronous-unwind-tables -fvisibility=hidden -fshort-enums -finline-small-functions -findirect-inlining -fno-common") 
+  #-flto -fuse-ld=gold # There is an issue with gold link when compiling on 
+  # Ubuntu 16. At that point gcc linker did not integrate the flto support 
+  # inside and it was a separate plugin that need to be added. Therefore, 
+  # this can be added when mlpack Azure CI moves toward Ubuntu 20.
```

Comment:
Was it worthwhile to use the gold linker in testing?  I remember doing some testing some time back and we found that it wasn't actually all that useful.  But maybe the situation is different for embedded devices?

Reply:
It has better support for `flto`optimization which help in reducing the size a little more, but it is very complicated to do that on Ubuntu 16, since the golden linker did not have support for `lto` inside. However, in recent version it is integrated in the golden linker. 
In the future (when mlpack moves to a recent version of Ubutnu) this will allow us to use the golden linker.

Reply:
Cool, thanks for the explanation. :+1: